ui/app/serializers/role.js

```diff
@@ -14,4 +14,18 @@ export default class RoleSerializer extends ApplicationSerializer {
     hash.PolicyIDs = hash.Policies.map((policy) => policy.Name);
     return super.normalize(typeHash, hash);
   }
+  serialize(snapshot, options) {
+    const hash = super.serialize(snapshot, options);
+    // required for update/PUT requests
+    if (snapshot.id) {
+      hash.ID = snapshot.id;
+    }
+    hash.Policies = hash.PolicyIDs.map((policy) => {
+      return {
+        Name: policy,
+      };
+    });
+    delete hash.PolicyIDs;
+    return hash;
+  }
```

Comment:
The way our API endpoints handle hasMany relationships (a list of IDs) and how the Ember app handles them (as a list of fully-fleshed-out objects) deviate from one another. This serializer layer handles that incongruence when we read/save.